### ISSUE TITLE
General multi language support

### ISSUE THREAD

**Issue by julkue** [drupal-graphql/graphql#616]
Basically I have a query to fetch e.g. a menu:

Query:

```graphql
query MainMenu {
  mainMenu: menuByName(name: "main") {
    ...MenuFieldsFragment
  }
}

fragment MenuFieldsFragment on Menu {
  name
  links {
    label
    expanded
    url {
      path
    }
  }
}
```

which can e.g. result in the following response:

```json
{
  "data": {
    "mainMenu": {
      "name": "Hauptnavigation",
      "links": [
        {
          "label": "Startseite",
          "expanded": true,
          "url": {
            "path": "/de"
          }
        },
        {
          "label": "Über uns",
          "expanded": true,
          "url": {
            "path": "/de/ueber-uns"
          }
        }
      ]
    }
  }
}
```

However, there's no argument to specify the expected response language:

![topic](https://user-images.githubusercontent.com/8310677/40781892-e1955994-64dd-11e8-819b-224b474c4896.png)

I know that there's a "translate" option for `url` to get the URL for a specific language, but that won't help me also translating the labels.

So I've searched a bit and found this comment from early 2017:
https://github.com/drupal-graphql/graphql/issues/30#issuecomment-297668090

> Yes, currently you can do that through your own custom schema. In the future, we will add this together with a submodule that will take over the job of generating a schema based on the Drupal data graph.

Is this still state of the art? If so, to be honest, a documentation for this would be highly appreciated.

I also found [this issue](https://github.com/drupal-graphql/graphql/issues/342) which started as the same issue that I have but then became an issue about normal user interface string translation, so there's no solution included.

Finally there's [this ticket](https://github.com/drupal-graphql/graphql/issues/600) saying:

> Language handling changed drastically in beta 7, basically because of the issues you also face

Beside of the fact that I don't know what "Beta 7" is referencing (because `"drupal/graphql": "3.x-dev"` is used in the [decoupled app](https://github.com/drupal-graphql/drupal-decoupled-app/blob/master/backend/composer.json#L18)), I think that the referenced documentation doesn't inlcude this use case:

> A couple of fields accept a "language" argument, and whenever this argument is filled explicitly, it's value will be inherited to subsequent occurrences. The route field will set this context implicitly from the paths language prefix.

This may solve the issue for `route`, but not for `menuByName` since there's no parent query with a language.

**So, to summarize: How can I specify a global language that is considered in the entire response?**

**Comment by julkue** [comment]
btw: I've also tried the `accept-language` header with no luck.

**Comment by das-peter** [comment]
I think, this requires a manual definition of the language.
Like this:

``` 
query MainMenu {
  mainMenu: menuByName(name: "main") {
    ...MenuFieldsFragment
  }
  languageInterfaceContext {
  	id
  	name
	}
}

fragment MenuFieldsFragment on Menu {
  name
  links {
    label
    expanded
    url {
      translate(language:EN) {
        path
      }
    }
  }
}
```

Changing the language in `translate(language:EN) ` will change the results - even the label. While language prefixes and such doesn't seem to have any effect.

**Comment by julkue** [comment]
@das-peter Thanks for your help. However, unfortunately your query results in the following for me:

```json
{
  "data": {
    "mainMenu": {
      "name": "Hauptnavigation",
      "links": [
        {
          "label": "Startseite",
          "expanded": true,
          "url": {
            "translate": {
              "path": "/en"
            }
          }
        },
        {
          "label": "Über uns",
          "expanded": true,
          "url": {
            "translate": {
              "path": "/en/node/3"
            }
          }
        }
      ]
    },
    "languageInterfaceContext": {
      "id": "de",
      "name": "German"
    }
  }
}
```

Labels are still not translated.

**Comment by fubhy** [comment]
We need to add a language input parameter for the menuByName field.

**Comment by fubhy** [comment]
Should be fixed by https://github.com/drupal-graphql/graphql/commit/cda8bc5f3d94cbf0bca4db06e8675e2e4a22f97e

**Comment by das-peter** [comment]
@fubhy Awesome! Thank you very much!!
